packages/flutter/test/material/app_bar_test.dart

```diff
@@ -926,9 +927,43 @@ void main() {
     expect(getAppBarWidget(appBarFinder).elevation, 0.0);
 
     await tester.pumpWidget(buildSliverAppBar(true));
+    expect(getAppBarWidget(appBarFinder).elevation, null);
```

Comment:
We'll need to add a little more to this test:

The main problem is that we haven't verified that SliverAppBar sets the default value of `elevation` to `4.0` when elevation and AppBarTheme.elevation are null: https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/app_bar.dart#L370.

Meaning, we also need the test to prove that the following priority order is preserved:
SliverAppBar.elevation > AppBarTheme.elevation > default value of 4. Currently, it just shows that `SliverAppBar.elevation`'s value is null, which doesn't tell us what elevation is actually painted by default.

So, we need to add more to the test to prove that the correct elevation is still painted by the AppBar onto `Material`, since it doesn't verify that the default. Here is how [an AppBarTheme test does it](https://github.com/flutter/flutter/blob/master/packages/flutter/test/material/app_bar_theme_test.dart#L27).

Reply:
Okay. I've modified the tests to check for an default elevation of 4.0 on the Material widget. Ideally, I would have referenced the constant inside _AppBarState, but because it's private, I had to hardcode the value of 4.0. This is also what the AppBarTheme test does that you linked to. I still have it check if the AppBar itself has an elevation of null, since that means that SliverAppBar allowed it to decide it's own elevation.